test/subdomainregistrar/TestRentalSubdomainRegistrar.js

```diff
@@ -0,0 +1,778 @@
+        ethers.constants.MaxUint256,
+      )
+      await expect(
+        SubdomainRegistrar3.register(
+          node,
+          'subname',
+          account2,
+          EMPTY_ADDRESS,
+          0,
+          86400 * 3, // greater than parent
+          [],
+        ),
+      ).to.be.revertedWith(`DurationTooLong("${namehash('test.eth')}")`)
+    })
+
+    it('should emit an event on registration', async () => {
+      const duration = 86400
+      await NameWrapper.setApprovalForAll(SubdomainRegistrar.address, true)
+      await Erc20WithAccount2.approve(
+        SubdomainRegistrar.address,
+        ethers.constants.MaxUint256,
+      )
+      const tx = SubdomainRegistrar2.register(
+        node,
+        'subname',
+        account2,
+        EMPTY_ADDRESS,
+        0,
+        duration,
+        [],
+      )
+      const expiry =
+        (await ethers.provider.getBlock('latest')).timestamp + duration
+
+      await expect(tx)
+        .to.emit(SubdomainRegistrar, 'NameRegistered')
+        .withArgs(namehash('subname.test.eth'), expiry)
+    })
+  })
+
+  describe('renew()', () => {
+    beforeEach(async () => {
+      await BaseRegistrar.register(labelhash('test'), account, 200000)
+      await BaseRegistrar.setApprovalForAll(NameWrapper.address, true)
+      await NameWrapper.wrapETH2LD(
+        'test',
+        account,
+        CANNOT_UNWRAP,
+        EMPTY_ADDRESS,
+      )
+      expect(await NameWrapper.ownerOf(node)).to.equal(account)
+      await SubdomainRegistrar.setupDomain(
+        node,
+        RentalPricer.address,
+        account,
+        true,
+      )
+      await NameWrapper.setApprovalForAll(SubdomainRegistrar.address, true)
+      await Erc20WithAccount2.approve(
+        SubdomainRegistrar.address,
+        ethers.constants.MaxUint256,
+      )
+      await SubdomainRegistrar2.register(
+        node,
+        'subname',
+        account2,
+        EMPTY_ADDRESS,
+        0,
+        86400,
+        [],
+      )
+
+      expect(await NameWrapper.ownerOf(subNode)).to.equal(account2)
+    })
+    it('should allow subdomains to be renewed', async () => {
+      const [owner, , expiry] = await NameWrapper.getData(
+        namehash('subname.test.eth'),
+      )
+      await SubdomainRegistrar2.renew(node, 'subname', 500)
+      const [, , expiry2] = await NameWrapper.getData(
+        namehash('subname.test.eth'),
+      )
+      expect(expiry2.toNumber()).to.equal(expiry.toNumber() + 500)
+    })
+
+    it('should allow subdomains to be renewed even with 0 registration fee', async () => {
+      const [, , expiry] = await NameWrapper.getData(
+        namehash('subname.test.eth'),
+      )
+      await SubdomainRegistrar2.renew(node, 'subname', 86400)
+      const [, , expiry2] = await NameWrapper.getData(
+        namehash('subname.test.eth'),
+      )
+
+      expect(expiry2.toNumber()).to.equal(expiry.toNumber() + 86400)
+    })
+
+    it('should revert if expiry is greater than parent', async () => {
```

Comment:
what is the parent expiry? If `84600`, it would be great to set that into a meaningful variable so that the test is easier to read